bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/status/IStatus.java

```diff
@@ -14,8 +14,8 @@
 package org.eclipse.ui.internal.findandreplace.status;
 
 /**
- * Interface for statuses that can occur while performing
- * Find/Replace-operations.
+ * Interface for statuses that can occur while performing find/replace
+ * operations.
  */
 public interface IStatus {
```

Comment:
Why one needs a new `IStatus` and can't use the Eclipse `IStatus` interface here?

Reply:
I think it's **really unfortunate** to reuse the name IStatus because that can only lead to unfortunate confusion and we already see that here.  When I saw the comment from the PR in my inbox, I thought, "Are you kidding my, they are doing to change IStatus?!"...

Reply:
This is only an internal status object for exchange of very specific information between find/replace business logic and UI, for which Eclipse's `IStatus` is neither required nor really appropriate. From a pure technical perspective, the name of this interface is `[...].findandreplace.IStatus`, as, of course, the package hierarchy also belongs to the name (c.f. types like `Event`).
But I totally see your point that the name clash is bad for something so essential as `IStatus` and that it makes sense to give it a more specific name. I will extend the PR to also adapt the name of that interface.

Reply:
@HeikoKlare why is `IStatus` (eclipse) not appropriate here? I must confess I have not reviewed the original in detail so its maybe a bit late now but I don't see this status being much different (except that you can visit childs, what could be either a static helper method or be added to `IStatus` (eclipse) as well...

Reply:
Eclipse's `IStatus` is a general concept to provide all kinds of status information about the outcome of an operation to be processed uniformly at different places. Here, we just need a very restricted set of specific status information that is only passed between the `FindReplaceLogic` and its consumers (i.e. find/replace UIs).
Using Eclipse's `IStatus` would result in worse design, as we have to encode information into "magic" integer values rather than providing a suitable API and as we would have to make the status objects provide messages, whereas we introduced these status objects to move the responsibility for generating a UI representation (such as a message) away from the business logic into the UI. At the same time, at least from my point of view there is no real benefit in using Eclipse's `IStatus` here. If you see a benefit in using Eclipse's `IStatus` that justifies the loss in specificity/design, please let me know and we can of course adapt that (everything here is internal anyway).

Reply:
I will assert again that we would not be having this discussion at all if it had a different name to begin with.  :scream:  I don't think we need to make this foot fit into a the IStatus shoe...

Reply:
My only concern here is that you have an interface and then check for specific implementations... that actually shows that the "clean api design" is actually not an api (maybe a facade) so probably you more want a search context (!) instead that allows to pass information you need around and then maybe has a Type enum...

Reply:
> My only concern here is that you have an interface and then check for specific implementations... that actually shows that the "clean api design" is actually not an api (maybe a facade) so probably you more want a search context (!) instead that allows to pass information you need around and then maybe has a Type enum...

That's exactly what this PR is supposed to fix 🙂

Reply:
If you look at the public API of the `IStatus` classes:

| **IStatus (search)** | **IStatus (eclipse)**           |
|----------------------|---------------------------------|
| wasSuccesfull        | isOk                            |
| isInputValid         | getSeverity() == IStatus.CANCEL |
| accept               | getChildren                     |

for me it looks like as the public api interface does matches 1:1 but IStatus(Eclipse) is a bit more options so for me it seems good to reuse an existing api/interface than inventing a new one.

Reply:
Please don't get me wrong: I am totally in favor of reusing implementations whenever possible, as it has a clear benefit (changes to that code, be they corrective, adaptive or perfective, apply to all usages). But this is not about code duplication but about an interface, which is designed to fit for a specific purpose. And, unfortunately, I do not understand the general idea of reusing an existing (rather generic) interface just because you can implement it in some way that makes it (syntactically) applicable in some scenario, because I do not see any benefit in that. Maybe you can clarify what you would expect from that reuse.
On the contrary, I see significant drawbacks in using a general-purpose rather than a tailored interface: the interface that we use here is _small_. It provides two methods that you could mimic with Eclipse's `IStatus` and provides one additional method (`accept()`, which is _not_ equal to `getChildren()`) that is not provided by Eclipse's `IStatus` at all. Eclipse's `IStatus` has 9 methods of which 7 would _not_ be used, so some default implementations for these methods without any meaning would be required. For the remaining two methods, you would need to document how the general interface is mapped to the specifics needed in this scenario, which in particular concerns the implicit meaning of `severity = IStatus.CANCEL`. You would have to place the documentation of current `isInputValid()` at some less reasonable point than at an according method that is called to retrieve that state.

And as I mentioned, Eclipse's `IStatus` assumes that a message is provided. The status objects for find/replace results are supposed to explicitly _not_ provide a message, as the status is only supposed to carry an abstract description of the status while the decision how to report it should be up to the UI. That is something that is infeasible for Eclipse's `IStatus` due to its open usage scope allowing all kinds of unknown subclasses, but it is reasonable in case all subclasses are known (like in this case).

Reply:
I've opened #1503 for the rename to keep concerns separate.